File: FresherAndroid/AndroidBasicCode/ManhPT14_Assignment6.2_Service/app/src/main/java/com/example/foregroundservicemusic/activity/MainActivity.java
package com.example.foregroundservicemusic.activity;

import android.Manifest;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import androidx.viewpager2.widget.ViewPager2;

import com.example.foregroundservicemusic.R;
import com.example.foregroundservicemusic.fragment.AlbumFragment;
import com.example.foregroundservicemusic.fragment.ArtistFragment;
import com.example.foregroundservicemusic.fragment.PlaylistFragment;
import com.example.foregroundservicemusic.service.MusicService;
import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        createNotificationChannel();
        requestNotificationPermission();
        setupViewPager();
    }

    private void createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            NotificationChannel channel = new NotificationChannel(
                    MusicService.CHANNEL_ID,
                    "Music Playback",
                    NotificationManager.IMPORTANCE_LOW
            );
            channel.setDescription("Foreground service for music playback");
            NotificationManager manager = getSystemService(NotificationManager.class);
            manager.createNotificationChannel(channel);
        }
    }

    private void requestNotificationPermission() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            if (checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS)
                    != PackageManager.PERMISSION_GRANTED) {
                ActivityCompat.requestPermissions(this,
                        new String[]{Manifest.permission.POST_NOTIFICATIONS}, 100);
            }
        }
    }

    private void setupViewPager() {
        ViewPager2 viewPager = findViewById(R.id.viewPager);
        TabLayout tabLayout = findViewById(R.id.tabLayout);

        viewPager.setAdapter(new FragmentStateAdapter(this) {
            @Override
            public Fragment createFragment(int position) {
                switch (position) {
                    case 1:
                        return new ArtistFragment();
                    case 2:
                        return new AlbumFragment();
                    default:
                        return new PlaylistFragment();
                }
            }

            @Override
            public int getItemCount() {
                return 3;
            }
        });

        String[] tabTitles = {"Playlists", "Artists", "Albums"};
        new TabLayoutMediator(tabLayout, viewPager,
                (tab, position) -> tab.setText(tabTitles[position])
        ).attach();
    }
}

File: FresherAndroid/AndroidAdvanced/submit/ManhPT14_advancedComponent/assignment1/app/src/main/java/com/example/galleryphoto/MainActivity.kt
package com.example.galleryphoto

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.viewpager2.widget.ViewPager2
import com.example.galleryphoto.adapter.ImagePagerAdapter
import com.example.galleryphoto.model.ImageItem

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)
        // sample images
        val images = listOf(
            ImageItem(resId = R.drawable.billgate),
            ImageItem(resId = R.drawable.elon),
            ImageItem(resId = R.drawable.messi),
            ImageItem(resId = R.drawable.ronaldo),
            ImageItem(resId = R.drawable.stevejob),
            ImageItem(resId = R.drawable.trinhvanquyet),
            ImageItem(resId = R.drawable.truonggiabinh)
        )

        val viewPager = findViewById<ViewPager2>(R.id.viewPager)
        val tvIndicator = findViewById<android.widget.TextView>(R.id.tvIndicator)

        viewPager.adapter = ImagePagerAdapter(images)

        // Cập nhật chỉ số trang khi vuốt
        updateIndicator(tvIndicator, 0, images.size)

        viewPager.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
            override fun onPageSelected(position: Int) {
                super.onPageSelected(position)
                updateIndicator(tvIndicator, position, images.size)
            }
        })
    }

    private fun updateIndicator(tv: android.widget.TextView, position: Int, total: Int) {
        tv.text = "Image ${position + 1}/$total"
    }
}